openquakeplatform/openquakeplatform/ript/static/ript/js/fragility.js

```diff
@@ -0,0 +1,345 @@
+    if (funcType == 'discr') {
+        imls = '';
+    } else if (funcType == 'cont') {
+        imls =
+            '<label> MinIML: </label>' +
+            '<input name="min-impls" class="ffsTable" type="text"><br>' +
+            '<label> MaxIML: </label>' +
+            '<input name="max-impls" class="ffsTable" type="text">';
+    }
+
+    // Create the fragility function set (ffs)
+    $('.ff_gid #tables').append(
+        '<div id="table'+ff_obj.tbl_idx+'" class="tables_gid table'+ff_obj.tbl_idx+'_id ffsTableDiv panel panel-default" '+
+            'data-gem-func-type="'+ funcType + '">' +
+          '<strong class="ffsTitle">' + format_name.toUpperCase() + '</strong><button name="destroy_table" class="destroyTable btn-danger btn">Remove</button><br>' +
+            '<div class="ffsForm" >' +
+                '<label> Function Id: </label>' +
+                '<input name="id" class="ffsTable" type="text"><br>' +
+                '<label> IMT: </label>' +
+                '<input name="imt" class="ffsTable" type="text" placeholder="PGA"><br>' +
+                '<label> Damage Limit: </label>' +
+                '<input name="no-damage-limit" class="ffsTable" type="text"><br>' +
+                imls +
+                '<br>' +
+            '</div>'+
+            '<div style="width: 45%; float: right;">' +
+              '<div name="tableDiv'+ff_obj.tbl_idx+'" class="theTable"></div><br>' +
+            (funcType == 'discr' ?
+              '<button id="new_row_add" style="clear: right; float: right; margin-top: 4px;" class="btn">Add Row</button><br>' :
+             '') +
+           '</div>' +
+         '</div>'
+    );
+
+    // force bootstrap style
+    $('.ff_gid .btn-danger').css({'background-color': '#da4f49'});
+
+
+    ////////////////////////////////
+    /// fragility Table Settings ///
+    ////////////////////////////////
+
+    var handson_params = {
+        colHeaders: header,
+        startCols: headerLength,
+        maxCols: headerLength,
+        startRows: limitStateLength,
+        colWidths: colWidth
+    }
+
+    if (funcType == 'discr') {
+        handson_params.rowHeaders = true;
+        handson_params.contextMenu = true;
+        handson_params.manualColumnResize = true;
+        handson_params.manualRowResize = true;
+    }
+    else if (funcType == 'cont') {
+        handson_params.rowHeaders = false;
+        handson_params.contextMenu = false;
+        handson_params.manualColumnResize = false;
+        handson_params.manualRowResize = false;
+        handson_params.maxRows = ff_obj.limitStates.length;
+        handson_params.columns = [
+            {readOnly: true,
+             className: "htLeft"},
+            {className: "htRight"},
+            {className: "htRight"}
+        ],
+
+        handson_params.cells = function(r,c, prop) {
+            var cellProperties = {};
+            if (c===0) cellProperties.readOnly = true;
+            return cellProperties;
+        }
```

Comment:
jshint gives me 2 warnings here:

* missing `;`

* Expected an assignment or function call and instead saw an expression